test/tm/generator.go

```diff
@@ -29,20 +31,41 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
-var (
-	infrastructureProviderConfigPath = flag.String("infrastructure-provider-config-filepath", "", "filepath to the provider specific infrastructure config")
-	controlplaneProviderConfigPath   = flag.String("controlplane-provider-config-filepath", "", "filepath to the provider specific controlplane config")
+const (
+	defaultNetworkVnetCIDR   = "10.250.0.0/16"
+	defaultNetworkWorkerCidr = "10.250.0.0/19"
+)
 
-	networkVnetCidr   = flag.String("network-vnet-cidr", "10.250.0.0/16", "vnet network cidr")
-	networkWorkerCidr = flag.String("network-worker-cidr", "10.250.0.0/19", "worker network cidr")
+type GeneratorConfig struct {
+	networkWorkerCidr                string
+	networkVnetCidr                  string
+	infrastructureProviderConfigPath string
+	controlplaneProviderConfigPath   string
+	zonedFlag                        string
+	zoned                            bool
+}
 
-	zoned = flag.Bool("zoned", false, "shoot uses multiple zones")
+var (
+	cfg    *GeneratorConfig
+	logger logr.Logger
 )
 
+func addFlags() {
+	cfg = &GeneratorConfig{}
+	flag.StringVar(&cfg.infrastructureProviderConfigPath, "infrastructure-provider-config-filepath", "", "filepath to the provider specific infrastructure config")
+	flag.StringVar(&cfg.controlplaneProviderConfigPath, "controlplane-provider-config-filepath", "", "filepath to the provider specific controlplane config")
+
+	flag.StringVar(&cfg.networkVnetCidr, "network-vnet-cidr", "10.250.0.0/16", "vnet network cidr")
```

Comment:
```suggestion
	flag.StringVar(&cfg.networkVnetCidr, "network-vnet-cidr", "", "vnet network cidr")
```

I think `validate()` already sets the default, so probably remove it here.